htdocs/core/class/html.formmail.class.php

```diff
@@ -1082,6 +1082,12 @@ public function getHtmlForTo()
 					$withtoselected = array_keys($tmparray);
 				}
 
+				// The function getFormMail is the only one that calls this one, so one uses the hook
+				// getFormMail to set $this->withtoselected via $object->withtoselected in the hook
+				if (!empty($this->withtoselected)) {
```

Comment:
Why don't you set the ->withto  ?  it seems it is the one dedicated to set  $withtoselected to your own value